copyright/copyright.js

```diff
@@ -0,0 +1,108 @@
+/* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
+/* Copyright (c) 2017 Mobify Research & Development Inc. All rights reserved. */
+/* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
+
+const glob = require('glob')
+const fs = require('fs')
+const path = require('path')
+
+const red = '\x1b[31m'
+const green = '\x1b[32m'
+const yellow = '\x1b[33m'
+const magenta = '\x1b[35m'
+const cyan = '\x1b[36m'
+const blackBG = '\x1b[40m'
+const defaultBG = '\x1b[49m'
+const defaultFG = '\x1b[39m'
+
+const args = process.argv.filter((arg) => {
+    return !/node|copyright/.test(arg)
+})
+
+const copyright = {
+    lintMode: false,
+    langs: {},
+    run() {
+        this.buildSupportedExtensions()
+        const processedGlobs = args.map((folder) => {
+            return new Promise((resolve) => {
+                glob(`${folder}`, (err, file) => {
+                    resolve(file[0])
+                })
+            })
+        })
+
+        Promise.all(processedGlobs).then((files) => {
+            const filesContainingHeader = files.filter((file) => {
+                const content = fs.readFileSync(file)
+                const hasCopyrightHeader = content.includes('Copyright (c)')
+                const ext = file.match(/\.[0-9a-z]+$/i)[0]
+
+                if (hasCopyrightHeader) {
+                    return true
+                } else {
+                    if (this.lintMode) {
+                        console.log(`${yellow}${file} ${red}missing copyright header`)
+                        return false
+                    } else {
+                        const newData = this.getHeaderText(ext) + content
+                        fs.writeFileSync(file, newData)
+                        console.log(`${green}Copyright header succesfully written into ${magenta}${file}`)
+                        return true
+                    }
+                }
```

Comment:
Having restructured things in this way, we don't actually need the `else` block; the `return` statement will exit the function.